src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -0,0 +1,391 @@
+}
+
+/*
+Function:
+EnumMonths
+
+Enumerates all the month names of the specified context and width, invoking the callback function
+for each month name.
+*/
+bool EnumMonths(
+	DateFormatSymbols& dateFormatSymbols,
+	DateFormatSymbols::DtContextType dtContext,
+	DateFormatSymbols::DtWidthType dtWidth,
+	EnumCalendarInfoCallback callback,
+	void* context)
+{
+	int32_t monthsCount;
+	const UnicodeString* monthNames = dateFormatSymbols.getMonths(monthsCount, dtContext, dtWidth);
+	return EnumCalendarArray(monthNames, monthsCount, callback, context);
+}
+
+/*
+Function:
+EnumCalendarInfo
+
+Retrieves a collection of calendar string data specified by the locale, calendar, and data type.
+Allows for a collection of calendar string data to be retrieved by invoking
+the callback for each value in the collection.
+The context parameter is passed through to the callback along with each string.
+*/
+extern "C" bool EnumCalendarInfo(
+	EnumCalendarInfoCallback callback,
+	UChar* localeName,
+	CalendarId calendarId,
+	CalenderDataType dataType,
+	void* context)
+{
+	UErrorCode err = U_ZERO_ERROR;
+
+	Locale locale = GetLocale(localeName);
+	if (locale.isBogus())
+		return false;
+
+	if (dataType == CalenderDataType::ShortDates)
+	{
+		return InvokeCallbackForDateTimePattern(locale, "Mdyyyy", callback, context);
+	}
+
+	if (dataType == CalenderDataType::LongDates)
+	{
+		// TODO: need to replace the "EEEE"s with "dddd"s for .net
+		// Also, "LLLL"s to "MMMM"s
+		return InvokeCallbackForDateTimePattern(locale, "eeeeMMMMddyyyy", callback, context);
+	}
+
+	if (dataType == CalenderDataType::YearMonths)
+	{
+		return InvokeCallbackForDateTimePattern(locale, "yyyyMMMM", callback, context);
+	}
+
+	DateFormatSymbols dateFormatSymbols(locale, GetCalendarName(calendarId), err);
+	if (!U_SUCCESS(err))
+		return false;
+
+	if (dataType == CalenderDataType::DayNames)
+	{
+		return EnumWeekdays(
+			dateFormatSymbols,
+			DateFormatSymbols::STANDALONE,
+			DateFormatSymbols::WIDE,
+			callback,
+			context);
+	}
+
+	if (dataType == CalenderDataType::AbbrevDayNames)
+	{
+		return EnumWeekdays(
+			dateFormatSymbols,
+			DateFormatSymbols::STANDALONE,
+			DateFormatSymbols::ABBREVIATED,
+			callback,
+			context);
+	}
+
+	if (dataType == CalenderDataType::MonthNames)
+	{
+		return EnumMonths(
+			dateFormatSymbols,
+			DateFormatSymbols::STANDALONE,
+			DateFormatSymbols::WIDE,
+			callback,
+			context);
+	}
+
+	if (dataType == CalenderDataType::AbbrevMonthNames)
+	{
+		return EnumMonths(
+			dateFormatSymbols,
+			DateFormatSymbols::STANDALONE,
+			DateFormatSymbols::ABBREVIATED,
+			callback,
+			context);
+	}
+
+	if (dataType == CalenderDataType::SuperShortDayNames)
+	{
+		return EnumWeekdays(
+			dateFormatSymbols,
+			DateFormatSymbols::STANDALONE,
+			DateFormatSymbols::SHORT,
+			callback,
+			context);
+	}
+
+	if (dataType == CalenderDataType::MonthGenitiveNames)
+	{
+		return EnumMonths(
+			dateFormatSymbols,
+			DateFormatSymbols::FORMAT,
```

Comment:
Yes, from everything that I can tell.  The [CLDR doc](http://cldr.unicode.org/translation/date-time) (see "Stand-Alone vs. Format Styles") that I read says that STANDALONE is used when the month/day is by itself, and FORMAT is used when the month is paired with a day.
I tested this using the pl-PL culture, and the output I got on Windows and Linux were the same for MonthNames & MonthGenitiveNames.